service-jehlomat/src/main/kotlin/api/SyringeApi.kt

```diff
@@ -34,6 +34,42 @@ fun Route.syringeApi(database: DatabaseService, mailer: MailerService): Route {
             call.respond(HttpStatusCode.OK, SyringeFilterResponse(filteredSyringes, pageInfo))
         }
 
+        authenticate(JWT_CONFIG_NAME) {
+
+            post("export") {
+                val loggedInUser = jwtManager.getLoggedInUser(call, database)
+                val user = database.selectUserById(loggedInUser.id)
+                val organization = user?.let { it1 -> database.selectOrganizationById(it1.organizationId) }
+                val roles = PermissionService.determineRoles(loggedInUser, organization)
+                if (!roles.contains(Role.OrgAdmin) && !roles.contains(Role.SuperAdmin)) {
+                    call.respond(HttpStatusCode.Forbidden, "Only admin or superadmin can export database")
+                    return@post
+                }
+
+                val request = call.receive<SyringeFilter>()
+
+                val organizationId = if (roles.contains(Role.SuperAdmin)) {
+                    null
+                } else {
+                    database.selectUserById(loggedInUser.id)?.organizationId
+                }
+
+                val output = database.selectSyringes(request, organizationId).map {it.toString()}.joinToString ("\n")
```

Comment:
I'm worried about this approach. It stores the whole selected row set in the memory, which could be potentially quite huge. I would either add some limitation for the filter, e.g. maximum time interval, or change it to streaming.

Reply:
I would probably need an explanation please:
You mean return stream from selectSyringes or convert output to stream, because it is string just string now that I am not able convert to a stream.
Thanks.

Reply:
The selectSyringes would have to be a stream, which would be continuously processed in the SyringeApi.kt and gradually sending to the output. But as I said, I don't know if our SQL framework allows to stream a result set. Otherwise it would be pointless.

So maybe the best approach would be just add some limits for the exported data, like maximum time interval.

Reply:
Please see: https://github.com/cesko-digital/jehlomat/pull/92/commits/28fc57251973e7d8039e44f887be22f7d379fba8